libs/facilities/frontend/view/src/lib/components/detail/detail.page.ts

```diff
@@ -162,21 +167,69 @@ export class XdDetailPage implements OnInit {
 			},
 		],
 	};
-	protected readonly envChart: Signal<EChartsOption | undefined> = computed(() => {
-		const envData = this.envData();
-		if (!envData) return undefined;
 
-		const envChart = {
-			...this.envOptions,
-		};
+    protected readonly envChart: Signal<EChartsOption | undefined> = computed(() => {
+        const envData = this.envData();
+        if (!envData) return undefined;
 
-		if (!envChart.series || !(envChart.series instanceof Array)) return undefined;
+        const envChart = {
+            ...this.envOptions,
+        };
+
+        if (!envChart.series || !(envChart.series instanceof Array)) return undefined;
+
+        envChart.series[0].data = envData.map((item) => [ item.time, item['Temperature'] ]);
+        envChart.series[1].data = envData.map((item) => [ item.time, item['Humidity'] ]);
+        envChart.series[2].data = envData.map((item) => [ item.time, item['Pressure'] ]);
+        return envChart;
+    });
+
+    protected readonly metricsChart: Signal<EChartsOption | undefined> = computed(() => {
+        const facility = this.facility();
+        if (!facility) return undefined;
+
+        const metrics = facility.metrics;
+
+        const xAxisData = map(metrics, item => PUMP_METRICS_FULL_NAME_MAP[item.name].replace(/ /g, '\n').trim());
+        const seriesKeys = $enum(EMetricsCategory).getValues();
+
+        const seriesData = map(seriesKeys, (key) => ({
+            type: 'bar',
+            name: METRIC_CATEGORY_COLOR_INFORMATION[key].abbreviation,
+            data: map(metrics, item => item[key]),
+            emphasis: { focus: 'series' },
+            itemStyle: { color: METRIC_CATEGORY_COLOR_INFORMATION[key].color },
+        }));
+
+        return {
+            tooltip: {
+                trigger: 'axis'
+            },
+            legend: {
+                top: 30,
+                left: 80,
+            },
+            grid: {
+                top: 80,
+            },
+            title: {
+                text: 'Pump Metrics',
+                left: 'center',
+            },
+            xAxis: {
+                type: 'category',
+                data: xAxisData,
+                nameLocation: 'middle',
```

Comment:
I noticed a problem with the labels. Depending on the screen size they sometimes disappear because the long text does not fit and then they just hide it.
![grafik](https://github.com/amosproj/amos2024ss01-xcelerator-demo-app/assets/166606862/c529e598-fabd-4e19-8ced-3232a3ff92fa)

To avoid this you can add
```
    axisLabel: {
          interval: 0,
      }
```
I got it from Stackoverflow
https://stackoverflow.com/questions/44076557/echarts-how-to-show-all-axis-labels